server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByEngineDataSet.java

```diff
@@ -70,36 +77,69 @@ public GroupByEngineDataSet(QueryContext context, GroupByTimePlan groupByTimePla
       long intervalNum = (long) Math.ceil(queryRange / (double) slidingStep);
       curStartTime = slidingStep * (intervalNum - 1) + startTime;
     }
-    curEndTime = Math.min(curStartTime + interval, endTime);
+
+    //if is group by months interval and sliding step are calculated in ms by * 30 * 86400_000L
+    //now converting them back to integer months
+    if (isIntervalByMonth) {
+      interval = interval / 30 / 86400_000L;
+      curEndTime = Math.min(curStartTime + calcIntervalByMonth(interval, curStartTime), endTime);
```

Comment:
Please add some comments about the implement. You could list some example to explain it.

Reply:
added. Thank you!